python/lsst/ts/mtreflector/controller.py

```diff
@@ -0,0 +1,437 @@
+            )
+        except asyncio.CancelledError:
+            self.log.info(
+                "run_in_thread cancelled while running "
+                f"blocking function {self._blocking_actuate_mtreflector}."
+            )
+        except Exception:
+            self.log.exception(
+                f"Blocking function {self._blocking_actuate_mtreflector} failed."
+            )
+            raise
+
+    def _blocking_actuate_mtreflector(self, actuate: MTReflectorStatus) -> None:
+        """Tell labjack to actuate reflector.
+
+        Parameters
+        ----------
+        actuate : `MTReflectorStatus`
+            open/close to actuate the reflector
+
+        Raises
+        ------
+        RuntimeError
+            If the Labjack cannot connect
+            If the Labjack reports back an error
+        TypeError
+            If actuate includes an item that is not OPEN or CLOSE
+        """
+        # confirm we have valid reflector values
+        if actuate not in [
+            MTReflectorStatus.OPEN,
+            MTReflectorStatus.CLOSE,
+        ]:
+            raise TypeError(f"{actuate} is an invalid state to set the reflector to")
+
+        # connect to labjack if we currently are disconnected
+        if self.handle is None:
+            try:
+                self.log.info("Attempting to connect to Labjack...")
+                self._blocking_connect()
+            except RuntimeError:
+                raise RuntimeError("Labjack can't connect")
+
+        self.log.info(f"Switching Reflector to {actuate}")
+
+        # form list of addresses and values to write
+        addresses = [self.labjack_item.address()]
+        values = [self.labjack_item.value(actuate)]
+        try:
+            ljm.eWriteAddresses(
+                self.handle,
+                len(addresses),
+                addresses,
+                [ljm.constants.UINT16 for _ in range(len(addresses))],
+                values,
+            )
+        except ljm.LJMError as ljm_error:
+            # Set up log string
+            error_code = ljm_error.errorCode
+            error_string = str(ljm_error)
+            log_string = str(
+                f"Labjack reported error#{error_code} during eWriteAddress"
+                f"in actuate_mtreflector, dumping values: "
+                f"actuate state: {actuate} "
+                f"handle: {self.handle} addresses: {addresses} "
+                f"data_type: {ljm.constants.UINT16} values_written: {values} "
+                f"ljm_error_string: {error_string}"
+            )
+
+            # If error then raise except else its a warning so continue
+            if error_code > ljm.errorcodes.WARNINGS_END:
+                self.log.exception(log_string)
+                raise RuntimeError(f"ljm reported error, see log: {log_string}")
+            self.log.warning(log_string)
+
+        # Update state
+        self.labjack_item.status = actuate
+
+    def _blocking_connect(self) -> None:
+        """
```

Comment:
Please remove the newline character and start the doc string immediately following the `"""`.